### ISSUE TITLE
Storage options passed to unexpected backend

### ISSUE THREAD

**Issue by ian-r-rose** [intake/filesystem_spec#388]
I'm looking into chaining caching with HTTP/S3 access as part of intake/intake_geopandas#15, and found some unexpected behavior with with chaining. My expectation of `**storage_options` was that each dict would be passed as kwargs to the relevant sub-filesystem. The problem I ran into is this: suppose I have a set of paths, some of which are intended to be cached and some of which are not. In that case, the `HTTPFileSystem` without caching was passed the caching options, and it passed those options as kwargs to `requests`, which failed:

```python
import fsspec

path = "https://github.com/intake/filesystem_spec/raw/master/README.md"

# Succeeds
with fsspec.open("simplecache::" + path, simplecache={"cache_storage": "/tmp/testcache"}) as f:
    print(f.read())

# Fails as simplecache is passed to requests
with fsspec.open(path, simplecache={"cache_storage": "/tmp/testcache"}) as f:
    print(f.read())
```

I suppose I had expected storage options for filesystems not involved in a given path to be ignored. Is this behavior expected?

**Comment by martindurant** [comment]
In the second, failing case, I can't immediately think of a way to tell that the passed dict is not a valid kwarg for the actual http filesystem. Many of the implementations can take a large variety of arguments.

**Comment by martindurant** [comment]
I wonder whether we can guarantee that no real backend kwarg conflicts with the set of known protocol strings?

**Comment by ian-r-rose** [comment]
> I wonder whether we can guarantee that no real backend kwarg conflicts with the set of known protocol strings?

Right, this is close to the behavior I expected: that `simplecache={ **simplecache_kwargs }` would only be passed to the simple cache backend, `http={ **http_kwargs }` would only be passed to the HTTP backend, etc.

**Comment by martindurant** [comment]
The extraction of kwargs per backend was en extra convenience that came when chained URLs were first introduced, else you would do
```
fsspec.open("simplecache::..", target_protocol="http", target_kwargs={'http_arg1': value})
```
which does still work now - but is not helpful to your case. The original intent was that all kwargs are passed to the backend.

Picking out the kwargs is done only by looking at the protocols of the URL, and without reference to the known/loaded backends. Thinking about this might help also with solving #384.

**Comment by ian-r-rose** [comment]
> Picking out the kwargs is done only by looking at the protocols of the URL, and without reference to the known/loaded backends. Thinking about this might help also with solving #384.

So currently it passes all the `storage_options` to each backend, *unless* the name of a kwarg matches one of the backends, and then it passes those to only that backend? It's coherent, but kind of easy for a user (me) to get wrong.

My hope was to provide some sensible defaults for some backends, which could be applied regardless of whether the backend is actually used for a given fsspec-path. A downstream workaround could be to pre-tokenize the URL and only pick out the relevant kwargs from my defaults.

**Comment by martindurant** [comment]
Yes, there is an issue around for being to set defaults for each backend - that would presumably be the one instantiated in the case of no arguments at all. 

> it passes all the storage_options to each backend

No, only the outer-most one, which would be the first, left-to-right, in the URL

**Comment by martindurant** [comment]
cc #339

**Comment by martindurant** [comment]
#453 would solve this?
